motionplan/tpSpaceRRT_test.go

```diff
@@ -303,3 +303,169 @@ func TestTPsmoothing(t *testing.T) {
+	opt.GoalThreshold = 30.
+
+	test.That(t, err, test.ShouldBeNil)
+	sf, err := newSolverFrame(fs, ackermanFrame.Name(), referenceframe.World, nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	mp, err := newTPSpaceMotionPlanner(ackermanFrame, rand.New(rand.NewSource(42)), logger, opt)
+	test.That(t, err, test.ShouldBeNil)
+	tp, _ := mp.(*tpSpaceRRTMotionPlanner)
+
+	plan, err := tp.plan(context.Background(), goalPos, nil)
+	test.That(t, err, test.ShouldBeNil)
+	planAsInputs := nodesToInputs(plan)
+	test.That(t, err, test.ShouldBeNil)
+	steps := []map[string][]referenceframe.Input{}
+	for _, resultSlice := range planAsInputs {
+		stepMap := sf.sliceToMap(resultSlice)
+		steps = append(steps, stepMap)
+	}
+
+	startPose := spatialmath.NewPoseFromPoint(r3.Vector{0, 0, 0})
+	errorState := startPose
+
+	t.Run("base case - validate plan without obstacles", func(t *testing.T) {
+		err := CheckPlan(ackermanFrame, steps, nil, fs, startPose, errorState)
+		test.That(t, err, test.ShouldBeNil)
+	})
+
+	t.Run("obstacles blocking path", func(t *testing.T) {
+		// create obstacle blocking path
+		obstacle, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{1000, 0, 0}), r3.Vector{20, 5000, 1}, "")
+		test.That(t, err, test.ShouldBeNil)
+
+		geoms := []spatialmath.Geometry{obstacle}
+		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)}
+
+		worldState, err := referenceframe.NewWorldState(gifs, nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = CheckPlan(ackermanFrame, steps, worldState, fs, startPose, errorState)
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("obstacles blocking path2", func(t *testing.T) {
+		// create obstacle blocking path
+		obstacle, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2000, 0, 0}), r3.Vector{20, 2000, 1}, "")
+		test.That(t, err, test.ShouldBeNil)
+
+		geoms := []spatialmath.Geometry{obstacle}
+		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)}
+
+		worldState, err := referenceframe.NewWorldState(gifs, nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = CheckPlan(ackermanFrame, steps, worldState, fs, startPose, errorState)
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("obstacles blocking path2.5", func(t *testing.T) {
+		// create obstacle blocking path
+		obstacle, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2500, 0, 0}), r3.Vector{20, 2000, 1}, "")
+		test.That(t, err, test.ShouldBeNil)
+
+		geoms := []spatialmath.Geometry{obstacle}
+		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)}
+
+		worldState, err := referenceframe.NewWorldState(gifs, nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = CheckPlan(ackermanFrame, steps, worldState, fs, startPose, errorState)
+		test.That(t, err, test.ShouldNotBeNil)
+	})
+
+	t.Run("obstacles blocking path3", func(t *testing.T) {
```

Comment:
It also looks like there are 4 test cases here that do a redundant thing and don't actually test anything different; they are just putting the obstacle at different places along the same plan.  Similarly, I'm not sure how the tests below are exercising different code paths

Also, do we test anything dealing with adding an offset from the plan?

Reply:
yep, I added a test in `tpSpaceRRT_test.go` called `"with offset"`